metals/src/main/scala/scala/meta/internal/metals/BuildTargets.scala

```diff
@@ -240,6 +240,17 @@ final class BuildTargets private (
     }
   }
 
+  def findBuildTargetForTwirlFile(twirlFile: AbsolutePath): Option[BuildTargetIdentifier] = {
+    data.iterator
+      .flatMap(data => data.allScala)
+      .filter { target =>
+        val baseDirectory = target.info.baseDirectory.toAbsolutePath
+        twirlFile.startWith(baseDirectory)
+      }
+      .headOption
```

Comment:
You should test if in some cases we can have several build targets that contains the Twirl file. (You can get that info by adding some logs with `scribe.info` or `scribe.warn`, publish locally, and use in your projects).

If we can have several build targets, how to find the best one?